pkg/cmd/server/origin/master.go

```diff
@@ -305,14 +296,30 @@ func (c *MasterConfig) InstallAPI(container *restful.Container) []string {
 	}
 
 	var root *restful.WebService
+	installed := 0
 	for _, svc := range container.RegisteredWebServices() {
 		switch svc.RootPath() {
 		case "/":
 			root = svc
 		case OpenShiftAPIPrefixV1Beta1:
 			svc.Doc("OpenShift REST API, version v1beta1").ApiVersion("v1beta1")
+
+			// add the current user filter
+			// TODO: factor this better
+			filter := currentUserContextFilter(c.getRequestsToUsers())
+			routes := svc.Routes()
+			for i := range routes {
+				route := &routes[i]
+				if route.Method == "GET" && (route.Path == OpenShiftAPIPrefixV1Beta1+"/ns/{namespace}/users/{name}" || route.Path == OpenShiftAPIPrefixV1Beta1+"/users/{name}") {
```

Comment:
This is adding a filter in front of getting any user that looks for ~ and transforms it.